gen/rules/src/main/scala/clue/gen/QueryGen.scala

```diff
@@ -180,7 +241,28 @@ trait QueryGen extends Generator {
       nameOverride:   Option[String] = none
     ): ClassAccumulator =
       currentAlgebra match {
-        case Select(name, _, child) =>
+        case Select(name, _, Select(fieldName, _, _)) if fieldName.startsWith("subquery") =>
+          val param = MetaTypes
+            .get(name)
+            .orElse(currentType.flatMap(_.field(name)))
+            .fold(
+              throw new Exception(
+                s"Could not resolve type for field [$name] - Is this a valid field present in the schema?"
+              )
+            ) { nextType =>
+              val i = fieldName.substring("subquery".length).toInt
+
+              ClassParam.fromGrackleType(
+                name,
+                nextType.dealias,
+                isInput = false,
+                typeOverride =
+                  Some(Type.Select(subqueries(i).asInstanceOf[Term.Ref], Type.Name("Data")))
+              )
```

Comment:
Most of the hacking is here. A synthetic "subqueryN" field is used to identify which subquery this is.

Also should revisit that `asInstanceOf` ... :)